src/components/query-viewer/utils.ts

```diff
@@ -0,0 +1,218 @@
+      componentItems.push(data.Name);
+    }
+  });
+  return componentItems;
+}
+
+function loadContextItems(
+  node: Node,
+  excludedItems: string[]
+): QueryViewerEventFilters[] | QueryViewerContextElement[] {
+  const items: QueryViewerContextElement[] = [];
+  if (Node === null) {
+    return null;
+  }
+  let itemIndex = -1;
+  node.childNodes.forEach(child => {
+    if (child.nodeName !== "ITEM") {
+      return null;
+    }
+    const itemName = (child as Element).getAttribute("name");
+    if (excludedItems.indexOf(itemName) < 0) {
+      itemIndex++;
+      items[itemIndex] = {
+        Name: itemName,
+        Values: []
+      };
+      let valueIndex = -1;
+      // Seek VALUES node
+      let valuesNode = child; // ITEM
+      child.childNodes.forEach(secondChild => {
+        if (secondChild.nodeName === "VALUES") {
+          valuesNode = secondChild;
+          //   break;
+        }
+      });
+      // Seek VALUE nodes
+      valuesNode.childNodes.forEach(valueChild => {
+        if (valueChild.nodeName === "VALUE") {
+          valueIndex++;
+          items[itemIndex].Values[valueIndex] =
+            valueChild.firstChild != null
+              ? valueChild.firstChild.nodeValue
+              : "";
+        }
+      });
+    }
+  });
+
+  return items;
+}
+
+export function itemExpandCollapsePivotTableEvent(
+  qViewer: any,
+  data: string,
+  isCollapse: boolean
+): QueryViewerItemExpandAndCollapseData {
+  //   if (IsQueryObjectPreview()) {
+  //     window.external.SendText(qViewer.ControlName, data);
+  //   }
+  if (
+    (qViewer.ItemExpand && !isCollapse) ||
+    (qViewer.ItemCollapse && isCollapse)
+  ) {
+    const eventData: QueryViewerItemExpandAndCollapseData = isCollapse
+      ? qViewer.ItemCollapseData
+      : qViewer.ItemExpandData;
+    const xmlDoc = xmlToDocument(data);
+    const dataItemNode = selectXPathNode(xmlDoc, "/DATA/ITEM");
+    const expandedValuesNode = selectXPathNode(
+      xmlDoc,
+      "/DATA/CONTEXT/EXPANDEDVALUES"
+    );
+    eventData.Name = (dataItemNode as Element).getAttribute("name");
+    eventData.Value = dataItemNode.firstChild.nodeValue;
+    eventData.ExpandedValues = [{ Item: "" }];
+    let valueIndex = -1;
+    expandedValuesNode.childNodes.forEach(child => {
+      if (child.nodeName !== "VALUE") {
+        return null;
+      }
+      valueIndex++;
+      eventData.ExpandedValues[valueIndex] = {
+        Item: child.firstChild.nodeValue
+      };
+    });
+
+    return eventData;
+  } else {
+    return null;
+  }
+}
+
+const getAxisType = (axis: string) => {
+  let axisType: QueryViewerAxisType = QueryViewerAxisType.Default;
+  switch (axis) {
+    case "rows":
+      axisType = QueryViewerAxisType.Rows;
+      break;
+    case "columns":
+      axisType = QueryViewerAxisType.Columns;
+      break;
+    default:
+      axisType = QueryViewerAxisType.Pages;
+      break;
+  }
+  return axisType;
+};
+
+export function dragAndDropPivotTableEvent(
+  _qViewer: any,
```

Comment:
Same comment as before

Reply:
Yes, that's true. I don't know what is the type because Andrés sends me this value within the event's parameter. I asked Andrés about the type of this value, and upon receiving his response, I'll proceed to change it. Thanks!! :)